xarray/core/variable.py

```diff
@@ -1629,19 +1630,39 @@ def _unstack_once(
         else:
             dtype = self.dtype
 
-        data = np.full_like(
-            self.data,
-            fill_value=fill_value,
-            shape=new_shape,
-            dtype=dtype,
-        )
+        if sparse:
+            # unstacking a dense multitindexed array to a sparse array
+            from sparse import COO
+
+            codes = zip(*index.codes)
+            if reordered.ndim == 1:
+                indexes = codes
+            else:
+                sizes = itertools.product(*[range(s) for s in reordered.shape[:-1]])
+                tuple_indexes = itertools.product(sizes, codes)
+                indexes = map(lambda x: list(itertools.chain(*x)), tuple_indexes)  # type: ignore
+
+            data = COO(
+                coords=np.array(list(indexes)).T,
```

Comment:
Is the np.array safe for pint? Or does it not matter here?

Reply:
it does not matter because right now `sparse` will convert anyways. Creating `sparse` arrays from `pint` might still be valid, but it's something that requires some sort of agreement first (see [pydata/duck-array-discussion](https://github.com/pydata/duck-array-discussion))

Reply:
indexes is just ints and *should* not have units so yeah should be OK.